File: Source/Java/datalabs/task/cache/TaskDataCache.java
package datalabs.task.cache;

import java.lang.reflect.InvocationTargetException;
import java.util.ArrayList;
import java.util.Map;

import datalabs.parameter.Parameters;


public abstract class TaskDataCache {
    public static enum Direction {
        INPUT, OUTPUT
    }

    protected Map<String, String> parameters = null;

    public TaskDataCache(Map<String, String> parameters)
            throws IllegalAccessException, InstantiationException, InvocationTargetException, NoSuchMethodException {
        this.parameters = parameters;
    }

    public abstract ArrayList<byte[]> extractData();

    public abstract void loadData(ArrayList<byte[]> outputData);
}

File: Source/Kotlin/datalabs/task/cache/TaskDataCache.kt
package datalabs.task.cache;

import java.util.ArrayList;

import datalabs.etl.s3.S3FileExtractorTask;
import datalabs.etl.s3.S3FileLoaderTask;
import datalabs.task.TaskException;


abstract class TaskDataCache(parameters: Map<String, String>) {
    enum class Direction {
        INPUT, OUTPUT
    }

    protected var parameters: Map<String, String> = parameters

    abstract fun extractData(): ArrayList<ByteArray>

    abstract fun loadData(outputData: ArrayList<ByteArray>)
}
